src/main/java/com/vaadin/demo/component/splitlayout/SplitLayoutToggle.java

```diff
@@ -1,21 +1,61 @@
 package com.vaadin.demo.component.splitlayout;
 
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.button.ButtonVariant;
 import com.vaadin.flow.component.html.Div;
-import com.vaadin.flow.component.html.Span;
-import com.vaadin.flow.component.select.Select;
+import com.vaadin.flow.component.icon.Icon;
+import com.vaadin.flow.component.icon.VaadinIcon;
 import com.vaadin.flow.component.splitlayout.SplitLayout;
 import com.vaadin.flow.router.Route;
 import com.vaadin.demo.DemoExporter; // hidden-source-line
 
 @Route("split-layout-toggle")
 public class SplitLayoutToggle extends Div {
+    private boolean sidebarCollapsed;
+    private final Button button;
+    private final Div masterContainer;
+    private final Icon leftArrowIcon;
+    private final Icon rightArrowIcon;
+    private final DetailContent detailContent;
 
     public SplitLayoutToggle() {
         // tag::snippet[]
-        SplitLayout splitLayout = new SplitLayout(new Span("Side 1"), new Span("Side 2"));
-        add(splitLayout);
+        button = new Button();
+        masterContainer = new Div();
+        leftArrowIcon = VaadinIcon.ARROW_LEFT.create();
+        rightArrowIcon = VaadinIcon.ARROW_RIGHT.create();
+        detailContent = new DetailContent();
+        MasterContent masterContent = new MasterContent();
+
+        sidebarCollapsed = false;
+
+        button.addClickListener(event -> {
+            sidebarCollapsed = !sidebarCollapsed;
+            setStyles();
+        });
+        button.getElement().setAttribute("aria-label", "Expand/collapse sidebar");
+        button.addThemeVariants(ButtonVariant.LUMO_TERTIARY, ButtonVariant.LUMO_ICON);
```

Comment:
Setting the Lumo Icon theme may be redundant here. 

`setStyles()` calls `button.setIcon(...)` which automatically [adds](https://github.com/vaadin/flow-components/blob/65ccf0b22f24053d922e0db7c25d04752daf8ecb/vaadin-button-flow-parent/vaadin-button-flow/src/main/java/com/vaadin/flow/component/button/Button.java#L208) the Lumo Icon theme to the button if no other content is provided except the icon.

```suggestion
        button.addThemeVariants(ButtonVariant.LUMO_TERTIARY);
```